lib/network/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib }:
+let
+  inherit (import ./internal.nix { inherit lib; }) _ipv6;
+in
+{
+  ipv6 = {
+    /**
+      Creates an `IPv6Address` object from an IPv6 address as a string. If
+      the prefix length is omitted, it defaults to 64. The parser is limited
+      to the first two versions of IPv6 addresses addressed in RFC 4291.
+      The form "x:x:x:x:x:x:d.d.d.d" is not yet implemented. Addresses are
+      NOT compressed, so they are not always the same as the canonical text
+      representation of IPv6 addresses defined in RFC 5952.
+
+      # Type
+
+      ```
+      fromString :: String -> IPv6Address
+      ```
+
+      # Examples
+
+      ```nix
+      fromString "2001:DB8::ffff/32"
+      => {
+        address = "2001:db8:0:0:0:0:0:ffff";
+        prefixLength = 32;
+        firstAddress = "2001:db8:0:0:0:0:0:0";
+        lastAddress = "2001:db8:ffff:ffff:ffff:ffff:ffff:ffff";
+        nextAddress = "2001:db8:0:0:0:0:1:0";
+      }
+      ```
+
+      # Arguments
+
+      - [addr] An IPv6 address with optional prefix length.
+    */
+    fromString =
```

Comment:
Instead of modifying `fromString`, we should have something like `lib.ipv6.firstAddress` and `lib.ipv6.lastAddress`.

Reply:
Hmm, can you explain roughly how you want to use the library? From my point of view, I expect something like `fromString <addr>` and then the user takes whatever data they need from the returned attr set.

Reply:
I guess that's fair, esp. since if we implement it correctly the values are lazily evaluated, which means it doesn't really increase the cost. But I still think we should expose the function to end users because it's more ergonomic to call `lib.ipv6.lastAddress "f00::c0ffee/64"` vs `(lib.ipv6.fromString "f00::c0ffee/64").lastAddress`. There will also be functions in the future that aren't really incorporatable into `fromString`, for example imagine something like `lib.ipv6.arithmetic "f00::c0ffee/64" (-50)`